### PR TITLE
drivers: fix gpio_mcux_lpc_port_set_masked_raw

### LINKED ISSUE #29070: NXP LPC GPIO driver masked set does not use the mask
**Describe the bug**

The NXP LPC GPIO driver (drivers/gpio/gpio_mcux_lpc.c) seems to have a problem in function gpio_mcux_lpc_port_set_masked_raw(). The `mask` parameter has no effect, because writes to `PIN` register ignore the `MASK` register value. I think the `PIN` register should replaced with `MPIN` register which does respect the `MASK` register:

```
diff --git a/drivers/gpio/gpio_mcux_lpc.c b/drivers/gpio/gpio_mcux_lpc.c
index 57903f6704..1a6bb0b47c 100644
--- a/drivers/gpio/gpio_mcux_lpc.c
+++ b/drivers/gpio/gpio_mcux_lpc.c
@@ -143,7 +143,7 @@ static int gpio_mcux_lpc_port_set_masked_raw(const struct device *dev,
 
        /* Writing 0 allows R+W, 1 disables the pin */
        gpio_base->MASK[port] = ~mask;
-       gpio_base->PIN[port] = value;
+       gpio_base->MPIN[port] = value;
        /* Enable back the pins, user won't assume pins remain masked*/
        gpio_base->MASK[port] = 0U;
 
```

This is also indicated by LPC55S69 User manual and experimentally verified.

**To Reproduce**

Sample code to reproduce this could look something like
```
    struct device *dev = device_get_binding(DT_LABEL(DT_NODELABEL(gpio0)));
    gpio_config(dev, 0, GPIO_OUTPUT);
    gpio_config(dev, 1, GPIO_OUTPUT);
    gpio_port_set_masked_raw(dev, 1 << 0, 1 << 0);
    gpio_port_set_masked_raw(dev, 1 << 1, 1 << 1);
    u32_t value;
    gpio_port_get_raw(dev, &value);
    printk("%x\n", value);
```
The latter gpio_port_set_masked_raw() call would clear bit 0 set by the first call. The actual pins one can use depend on chip, board and pinmux config, the pins 0 and 1 are just examples.

### PR DESCRIPTION
**Pull request by agansari**
Fixes gpio_mcux_lpc_port_set_masked_raw function inside gpio_mcux_lpc
driver. Writing masked data is correctly done using device registers.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

Fixes #29070

### CODE REVIEW

**Review by pabigot — APPROVED** [review]
Haven't dug into the datasheet but I can believe this is right.

The existing test is inadequate.  Would you be interested in adding a commit to fix that?  I think this code:
```
        rc = gpio_port_set_masked_raw(dev, BIT(PIN_IN), 0);
        zassert_equal(rc, 0,
                      "set_masked_raw low failed");
        zassert_equal(raw_in(), true,
                      "set_masked_raw low affected other pins");
```
inserted at https://github.com/zephyrproject-rtos/zephyr/blob/4d0750920b52e0ce8a3e188d08a1769ff7ec527b/tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c#L185 would do it: it should detect that the attempt to set only the input pin to zero also affected the output pin.

That should make the test fail on the unpatched code.

**Comment by agansari** [comment]
@pabigot sure

> Would you be interested in adding a commit to fix that?

**Comment by agansari** [comment]
@pabigot I've added the test and ran it with and without the previous commit, the result is as expected, PASS with fix, "set_masked_raw low affected other pins" i.e. FAIL without the mask fix.

Hope this PR does not transform in a fix all non-working pin masks.

File: tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c [hunk 1]
```diff
@@ -183,6 +183,12 @@ static int bits_physical(void)
 	zassert_equal(raw_in(), true,
 		      "set_masked_raw high mismatch");
 
+	rc = gpio_port_set_masked_raw(dev, BIT(PIN_IN), 0);
+	zassert_equal(rc, 0,
+					"set_masked_raw low failed");
```
Comment (pabigot) [inline tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c:188]: Indentation seems wrong.

**Comment by pabigot** [comment]
> Hope this PR does not transform in a fix all non-working pin masks.

Thanks; if it does cause other platforms to break that's a good thing to know.

**Review by MaureenHelm — APPROVED** [review]

**Review by pabigot — CHANGES_REQUESTED** [review]
Please fix indentation in test case.

**Review by pabigot — APPROVED** [review]

**Review by galak — APPROVED** [review]

**Pull request merged**